src/models/embeddingsList.ts

```diff
@@ -0,0 +1,135 @@
+import {
+  Embedding,
+  EmbeddingsList as OpenAPIEmbeddingsList,
+  EmbeddingsListUsage,
+} from '../pinecone-generated-ts-fetch';
+import * as util from 'node:util';
+
+/* This class wraps the OpenAPI-generated EmbeddingsList interface so that an EmbeddingsList object acts like an Array.
+This class also customizes the output of an EmbeddingsList to improve UX.
+
+```typescript
+const embeddingsList = new EmbeddingsList('someEmbeddingModel', [embedding1, embedding2], usage);
+console.log(embeddingsList);
+>>> EmbeddingsList({
+  "model": "someEmbeddingModel",
+  "data": [
+    "values": [0.1, 0.2, ..., 0.5, 0.6],
+    "values": [0.6, 0.7, ..., 1.0, 1.1]
+  ],
+  "usage": {"totalTokens": 2}
+})
+```
+*/
+export class EmbeddingsList
+  extends Array<Embedding>
+  implements OpenAPIEmbeddingsList
+{
+  model: string;
+  data: Array<Embedding>;
+  usage: EmbeddingsListUsage;
+
+  constructor(
+    model: string,
+    data: Array<Embedding>,
+    usage: EmbeddingsListUsage
+  ) {
+    super(...data);
+    // Set the prototype explicitly to ensure the instance is of type EmbeddingsList
+    Object.setPrototypeOf(this, EmbeddingsList.prototype);
+    this.model = model;
+    this.data = data;
+    this.usage = usage;
+  }
+
+  /* Customize format of output. */
+  public toString(): string {
+    const truncatedData = this.truncateData();
+    const dataObject = truncatedData
+      .map((embedding) => {
+        if (typeof embedding === 'string') {
+          return `    ${embedding}`;
+        }
+        let embeddingObject = JSON.stringify(embedding, (key, value) =>
+          key === 'values' && Array.isArray(value) ? value : value
+        );
+        embeddingObject = embeddingObject.replace(/:/g, ': ');
+
+        // Format the embedding itself
+        const valuesArray =
+          embeddingObject.match(/"values": \[(.*?)\]/)?.[1] || '';
```

Comment:
Just so I understand how this works, if someone was to want the "original" toString output, is there a way for them to do that?

Reply:
I don't believe so, no.

Reply:
As long as someone can get to the original/underlying type that is good enough

Reply:
Cool :)